app/src/main/java/com/example/fittracker/mvp/models/MainActivityModel.java

```diff
@@ -0,0 +1,48 @@
+package com.example.fittracker.mvp.models;
+
+import android.os.AsyncTask;
+
+import com.example.fittracker.User;
+import com.example.fittracker.UserDao;
+import com.example.fittracker.UserRoomDatabase;
+import com.example.fittracker.mvp.contracts.LogInContract;
+
+import java.util.concurrent.ExecutionException;
+
+public class MainActivityModel implements LogInContract.Model {
+    private UserRoomDatabase db;
+    private UserDao userDao;
+
+    public MainActivityModel(UserRoomDatabase db) {
+
+        this.db = db;
+        userDao = db.userDao();
+    }
+
+    @Override
+    public boolean isValidLogIn(User userInput) {
+        try {
+            return new fetchAsyncTask(userDao).execute(userInput).get();
```

Comment:
Why do you use an asyncTask here? are you showing something meantime the validation is happening?

Reply:
Room demands that all database operations must be made in the background thread.